### PR TITLE
P4-2355 Add feature flag ability to the API

### PR DESCRIPTION
**Pull request by cwrw**
jira-ticket: https://dsdmoj.atlassian.net/browse/P4-2355

Add [Flipper](https://github.com/jnunemaker/flipper) gem for feature flagging. We are utilising the [ActiveRecord adapter](https://github.com/jnunemaker/flipper/tree/master/docs/active_record). This includes migrations for feature tables, as well as config. For optimisation, we are using the [memoizing middleware](https://github.com/jnunemaker/flipper/blob/master/docs/Optimization.md) to only make one adapter call per feature per request.

For now, I will avoid adding an API endpoint or UI for using a feature flag due to added complexity with managing authentication and permissions regarding access, and will only allow enabling/disabling through the rails console: 
```
pry(main)> Flipper.enabled?(:prefill)
  Flipper::Adapters::ActiveRecord::Gate Load (0.4ms)  SELECT "flipper_gates".* FROM "flipper_gates" WHERE "flipper_gates"."feature_key" = $1  [["feature_key", "prefill"]]
=> false
pry(main)> Flipper.enable(:prefill)
=> true
pry(main)> Flipper.enabled?(:prefill)
  Flipper::Adapters::ActiveRecord::Gate Load (0.3ms)  SELECT "flipper_gates".* FROM "flipper_gates" WHERE "flipper_gates"."feature_key" = $1  [["feature_key", "prefill"]]
=> true
pry(main)> Flipper.disable(:prefill)
=> true
pry(main)> Flipper.enabled?(:prefill)
  Flipper::Adapters::ActiveRecord::Gate Load (0.3ms)  SELECT "flipper_gates".* FROM "flipper_gates" WHERE "flipper_gates"."feature_key" = $1  [["feature_key", "prefill"]]
=> false
```

Features don't have to be predefined to work.
This will allow us to feature flag pre-filling for the Person escort record on creation.

### CODE REVIEW

File: config/initializers/flipper.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Flipper.configure do |config|
+  config.default do
+    adapter = Flipper::Adapters::ActiveRecord.new
+    flipper = Flipper.new(adapter)
+  end
+
+end
+
+Rails.application.config.middleware.use Flipper::Middleware::Memoizer
+
```
Comment (willfish) [inline config/initializers/flipper.rb:10]: Annoying comment of the day but do you mind removing the extra lines?

File: config/initializers/flipper.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Flipper.configure do |config|
+  config.default do
+    adapter = Flipper::Adapters::ActiveRecord.new
+    flipper = Flipper.new(adapter)
+  end
+
```
Comment (willfish) [inline config/initializers/flipper.rb:6]: And here

**Review by willfish — APPROVED** [review]
Love 🐬

File: db/schema.rb [hunk 3]
```diff
@@ -10,7 +10,11 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
+<<<<<<< HEAD
```
Comment (willfish) [inline db/schema.rb:13]: Oops :)

**Review by willfish — APPROVED** [review]

**Review by smoothcontract — APPROVED** [review]
LGTM - looking forward to making use of this 👍

**Pull request merged**